### PR TITLE
Release 0.18

### PR DESCRIPTION
**Pull request by amueller**
final 0.18 release

### CODE REVIEW

**Comment by jakirkham** [comment]
So I should probably fix this up so we have 2 NumPy builds right? I'll get on that.

**Comment by amueller** [comment]
@jakirkham sorry, I might not have had enough coffee ... ?

**Comment by jakirkham** [comment]
Oh we were having issues with running the tests on CircleCI due to build time. If we split it into 2 branches, we can have each supported NumPy built against in each branch and still have time to run tests.

xref: https://github.com/conda-forge/scikit-learn-feedstock/pull/7#issuecomment-247214451

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ package:
 
 source:
   fn: scikit-learn-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/s/scikit-learn/scikit-learn-{{ version }}.tar.gz
+  url: https://github.com/scikit-learn/scikit-learn/archive/{{ version }}.tar.gz
   md5: a2f8b877e6d99b1ed737144f5a478dfc
```
Comment (jakirkham) [inline recipe/meta.yaml]: Need to change this. Also we might as well switch to sha256 while we are at it.

**Comment by jakirkham** [comment]
Please see PRs ( https://github.com/conda-forge/scikit-learn-feedstock/pull/12 ) and ( https://github.com/conda-forge/scikit-learn-feedstock/pull/13 ). These split off NumPy 1.10.x builds from `master` into a separate branch and keep NumPy 1.11.x builds on `master`. This way we don't go over CI limits when running the build.

The same strategy has been used with SciPy to some success. Admittedly this increases the maintenance work a bit. However, at some point we can just drop 1.10.x support at our discretion. If people still need it, they can take on more of the effort for maintaining it.

Long term we would like CircleCI to support some sort of build parallelism, but that is a bit tricky ATM. So this is an effective workaround.

**Comment by amueller** [comment]
ah, ok. so you'll fix that?

**Comment by jakirkham** [comment]
Fix which part? 😄

The breaking up of the NumPy matrix is simply a matter of merging those PRs and then rebasing this change. Will need a second PR though for NumPy 1.10 though.

The CircleCI parallelism issue is more of a long term (though solvable) problem. See issue ( https://github.com/conda-forge/conda-smithy/issues/203 ).

**Comment by jakirkham** [comment]
Hmm...did the build time speed up in this release? It seems like it may have based on the CI build times I'm looking at for those 2 PRs. If so, maybe splitting out the NumPy builds is not necessary. Could you please try fixing the checksum and then we can see if this works without that change?

**Comment by jakirkham** [comment]
Yeah, I estimate the build time here as ~1hr well below the 2hr limit for CircleCI. Not sure why it is so low now compared to the RC.

**Comment by jakirkham** [comment]
Apparently we still need to split NumPy 1.10 and 1.11 builds. Went ahead and merged those PRs. Could you please update `master` and rebase this change on it?

**Comment by amueller** [comment]
done

**Comment by jakirkham** [comment]
Thanks. Also please add another PR against the branch [`master_np1.10`]( https://github.com/conda-forge/scikit-learn-feedstock/tree/master_np1.10 ).

**Comment by jakirkham** [comment]
FYI have come up with an implementation for CircleCI matrix builds with `conda-smithy` in PR ( https://github.com/conda-forge/conda-smithy/pull/532 ). This should allow us to reunify the different NumPy based branches here and simplify the workflow of updating this feedstock going forward.

**Pull request merged**